src/Calculator/GraphingSettings.xaml.cpp

```diff
@@ -0,0 +1,220 @@
+    {
+        m_ParentGrapher->TrigUnitMode = (int)Graphing::EvalTrigUnitMode::Grads;
+    }
+}
+
+void GraphingSettings::UpdateExtents(Platform::Object ^ sender)
+{
+    FrameworkElement ^ fe = (FrameworkElement ^) sender;
+    FrameworkElement ^ currentFocus = (FrameworkElement ^) FocusManager::GetFocusedElement();
+
+    bool thisControlHasFocus = (fe == currentFocus);
+    TextBox ^ tb = reinterpret_cast<TextBox ^>(sender);
+    bool widthChanged = false;
+    bool heightChanged = false;
+
+    try
+    {
+        double newVal = std::stod(tb->Text->Begin());
+        if (tb->Name == "x_min" && newVal != m_xMin)
+        {
+            widthChanged = true;
+            m_xMin = newVal;
+            m_ParentGrapher->XAxisMin = m_xMin;
+        }
+
+        if (tb->Name == "x_max" && newVal != m_xMax)
+        {
+            widthChanged = true;
+            m_xMax = newVal;
+            m_ParentGrapher->XAxisMax = m_xMax;
+        }
+
+        if (tb->Name == "y_min" && newVal != m_yMin)
+        {
+            heightChanged = true;
+            m_yMin = newVal;
+            m_ParentGrapher->YAxisMin = m_yMin;
+        }
+
+        if (tb->Name == "y_max" && newVal != m_yMax)
+        {
+            heightChanged = true;
+            m_yMax = newVal;
+            m_ParentGrapher->YAxisMax = m_yMax;
+        }
+
+        // Only the focused element get's to trigger the recompute
+
+        if (thisControlHasFocus)
+        {
+            if (m_preserveAspectRatio)
+            {
+                if (widthChanged)
+                {
+                    m_width = m_xMax - m_xMin;
+                    m_height = m_width / m_currentAspectRatio;
+                    double bounds = m_height / 2;
+                    m_yMin = -bounds;
+                    m_yMax = bounds;
+                    m_ParentGrapher->YAxisMin = m_yMin;
+                    m_ParentGrapher->YAxisMax = m_yMax;
+                }
+
+                if (heightChanged)
+                {
+                    m_height = m_yMax - m_yMin;
+                    m_width = m_height * m_currentAspectRatio;
+                    double bounds = m_width / 2;
+                    m_xMin = -bounds;
+                    m_xMax = bounds;
+                    m_ParentGrapher->XAxisMin = m_xMin;
+                    m_ParentGrapher->XAxisMax = m_xMax;
+                }
+
+                if (widthChanged || heightChanged)
+                {
+                    x_min->Text = m_xMin.ToString();
+                    x_max->Text = m_xMax.ToString();
+
+                    y_min->Text = m_yMin.ToString();
+                    y_max->Text = m_yMax.ToString();
+
+                    m_ParentGrapher->SetDisplayRanges(m_xMin, m_xMax, m_yMin, m_yMax);
+                }
+            }
+        }
+    }
+    catch (const std::exception&)
+    {
+    }
+}
+
+void CalculatorApp::Controls::GraphingSettings::OnLoaded(Platform::Object ^ sender, Windows::UI::Xaml::RoutedEventArgs ^ e)
```

Comment:
Is this called every time this UI is opened? If not then there is a risk that this only fires once in the apps lifetime and somehow goes out of sync with the underlying data source.